database/gorm/query_test.go

```diff
@@ -350,6 +361,46 @@ func TestQueryTestSuite(t *testing.T) {
 	assert.Nil(t, sqlserverPool.Purge(sqlserverResource))
 }
 
+func TestCustomConnection(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping tests of using docker")
+	}
+
+	mysqlDocker := NewMysqlDocker()
+	mysqlPool, mysqlResource, query, err := mysqlDocker.New()
+	if err != nil {
+		log.Fatalf("Init mysql error: %s", err)
+	}
+	postgresqlDocker := NewPostgresqlDocker()
+	postgresqlPool, postgresqlResource, _, err := postgresqlDocker.New()
+	if err != nil {
+		log.Fatalf("Init mysql error: %s", err)
+	}
+
+	mysqlDocker.MockConfig.On("Get", "database.connections.postgresql.read").Return(nil)
+	mysqlDocker.MockConfig.On("Get", "database.connections.postgresql.write").Return(nil)
+	mysqlDocker.MockConfig.On("GetString", "database.connections.postgresql.host").Return("localhost")
+	mysqlDocker.MockConfig.On("GetString", "database.connections.postgresql.username").Return(DbUser)
+	mysqlDocker.MockConfig.On("GetString", "database.connections.postgresql.password").Return(DbPassword)
+	mysqlDocker.MockConfig.On("GetString", "database.connections.postgresql.driver").Return(ormcontract.DriverPostgresql.String())
+	mysqlDocker.MockConfig.On("GetString", "database.connections.postgresql.database").Return("postgres")
+	mysqlDocker.MockConfig.On("GetString", "database.connections.postgresql.sslmode").Return("disable")
+	mysqlDocker.MockConfig.On("GetString", "database.connections.postgresql.timezone").Return("UTC")
+	mysqlDocker.MockConfig.On("GetString", "database.connections.postgresql.prefix").Return("")
+	mysqlDocker.MockConfig.On("GetBool", "database.connections.postgresql.singular").Return(false)
+	mysqlDocker.MockConfig.On("GetInt", "database.connections.postgresql.port").Return(cast.ToInt(postgresqlResource.GetPort("5432/tcp")))
+
+	product := Product{Name: "create_product"}
+	assert.Nil(t, query.Create(&product))
+	assert.True(t, product.ID > 0)
+
+	var product1 Product
+	assert.Nil(t, query.Where("name", "create_product").First(&product1))
+	assert.True(t, product1.ID > 0)
```

Comment:
S: To verify the conditions are applied.

```suggestion
	var product1 Product
	assert.Nil(t, query.Where("name", "create_product").First(&product1))
	assert.True(t, product1.ID > 0)
	
	var product2 Product
	assert.Nil(t, query.Where("name", "create_product1").First(&product2))
	assert.True(t, product2.ID == 0)
```